test/src/test/java/hudson/slaves/DummyCloudImpl.java

```diff
@@ -106,6 +126,7 @@ public Node call() throws Exception {
             
             System.out.println("launching slave");
             DumbSlave slave = rule.createSlave(label);
+            slave.getNodeProperties().addAll(nodeProperties);
```

Comment:
A `NodeProperty` has an owning `node` (backreference) so this is dangerous: if multiple nodes were provisioned, the later ones would clobber the backreference from the earlier ones.

You should either switch this to be a `Provider<NodeProperty>`, or throw an assertion error if the cloud ever tries to provision multiple nodes and you had specified some `nodeProperties`.